srv/modules/runners/populate.py

```diff
@@ -697,17 +698,10 @@ def public_cluster(self, networks):
 
         # second step, find cluster network by checking which network salt-master does not belong
         master_addrs = []
-        master_minion = None
-        cmd_result = local.cmd(self.search, 'pillar.get', ['master_minion'], tgt_type="compound")
-        for _, value in cmd_result.items():
-            master_minion = value
-            break
-        if not master_minion:
-            raise Exception("No master_minion found in pillar")
-        cmd_result = local.cmd(master_minion, 'grains.get', ['ipv4'], tgt_type="compound")
-        for _, addr_list in cmd_result.items():
-            master_addrs.extend([ipaddress.ip_address(u'{}'.format(addr))
-                                for addr in addr_list if not addr.startswith('127.')])
+        __opts__ = salt.config.minion_config('/etc/salt/minion')
```

Comment:
This is reading `/etc/salt/minion`...

Reply:
This is the assignment you normally expect when you need grains or modules on the minions alone.  I think this could be considered the default for the number of examples out there.